3rdParty/iresearch/core/search/ngram_similarity_filter.cpp

```diff
@@ -446,6 +451,10 @@ class ngram_similarity_query : public filter::prepared {
 
       // add iterator
       itrs.emplace_back(std::move(docs));
+
+      if (!validate(itrs.back())) {
```

Comment:
Now there is a validation for the postings in `execute_ngram_similarity`, but postings are also fetched in `execute_simple_disjunction`. The latter location still uses the postings unvalidated.

Additionally, should other filters be hardened as well?
```
3rdParty/iresearch/core/search/phrase_filter.cpp:      auto docs = reader->postings(*term_state.first, features);
3rdParty/iresearch/core/search/phrase_filter.cpp:        disjunction_t::adapter docs(reader->postings(*term_state->first, features),
3rdParty/iresearch/core/search/multiterm_query.cpp:    auto docs = reader->postings(*entry.cookie, features);
3rdParty/iresearch/core/search/term_query.cpp:  auto docs = reader->postings(*state->cookie, ord.features());
3rdParty/iresearch/core/search/same_position_filter.cpp:      auto docs = reader->postings(*term_state.cookie, features);
```

Reply:
@jsteemann it's ok, there is a validation in other places too. Different queries require different checks.